### PR TITLE
New UI that allow creating multiple layers from sources supporting single layer per request.

### PR DESCRIPTION
**Pull request by mdastous-bentley**
When a WMTS or ArcGIS feature source is added, a new dialog will appear and ask end-user to pick the features/sub-layers he wants to add:
image

A map-layer instance will be created for each selected feature/sub-layer:
image

### CODE REVIEW

**Comment by mdastous-bentley** [comment]
@aruniverse A new build of map-layers widget will be needed once this PR merge.

File: common/changes/@itwin/map-layers/geo-maplayers_singleSubLayer_2023-06-13-12-32.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/map-layers",
+      "comment": "New UI that allow creating multiple layers from sources supporting single layer per request.",
+      "type": "patch"
```
[the remark this answers is not in the document]
Comment (mdastous-bentley) [reply common/changes/@itwin/map-layers/geo-maplayers_singleSubLayer_2023-06-13-12-32.json]: minor I guess..

File: packages/itwin/map-layers/package.json [hunk 2]
```diff
@@ -94,13 +94,13 @@
     "typescript": "~4.4.0"
   },
   "peerDependencies": {
-    "@itwin/appui-abstract": "^4.0.0",
+    "@itwin/appui-abstract": "^4.0.6",
```
[the remark this answers is not in the document]
Comment (mdastous-bentley) [reply packages/itwin/map-layers/package.json:97]: No new API, but a fix from core-frontend package is needed in order to get the UI appear to work properly.

[the remark this answers is not in the document]
Comment (mdastous-bentley) [reply packages/itwin/map-layers/package.json]: reverted

File: packages/itwin/map-layers/src/test/MapUrlDialog.test.tsx [hunk 3]
```diff
@@ -145,14 +156,15 @@ describe("MapUrlDialog", () => {
 
   });
 
-  const mockModalUrlDialogOk = () => {
+  const mockModalUrlDialogOk = (_result?: SourceState) => {
   };
 
   it("renders", () => {
     const component = enzyme.mount(<MapUrlDialog activeViewport={viewportMock.object} isOverlay={false} onOkResult={mockModalUrlDialogOk} />);
+    // console.log(component.html())
```
Comment (eringram) [inline packages/itwin/map-layers/src/test/MapUrlDialog.test.tsx:164]: Leftover console.log (assuming from testing)?

Comment (mdastous-bentley) [reply packages/itwin/map-layers/src/test/MapUrlDialog.test.tsx]: removed.

**Review by eringram — APPROVED** [review]

**Review by aruniverse — APPROVED** [review]

**Pull request merged**